boa/src/syntax/parser/function/mod.rs

```diff
@@ -76,14 +74,19 @@ where
         loop {
             let mut rest_param = false;
 
+            let position = cursor.peek(0)?.ok_or(ParseError::AbruptEnd)?.span().start();
             let next_param = match cursor.peek(0)? {
                 Some(tok) if tok.kind() == &TokenKind::Punctuator(Punctuator::Spread) => {
                     rest_param = true;
                     FunctionRestParameter::new(self.allow_yield, self.allow_await).parse(cursor)?
                 }
                 _ => FormalParameter::new(self.allow_yield, self.allow_await).parse(cursor)?,
             };
+            if param_names.contains(next_param.name()) {
+                return Err(ParseError::general("duplicate parameter name", position));
+            }
 
+            param_names.insert(next_param.name().to_string());
```

Comment:
This is creating a `HashSet<String>`, right? Is it possible to do it with just `&str` or similar? It would be nice to minimize allocations.

Reply:
Ideally, at some point we want to intern all these strings, but for now this is not possible. The only optimization that could be done reasonably fast would be to change the `Stri g` for a `Box<str>`.

Reply:
I guess that still minimizes the space used by the HashSet...

Reply:
I'll change this to a `Box<str>`. I couldn't think of a better way to check for duplicate names (except maybe storing their hashes?).